tree/treeplayer/src/TMPWorkerTree.cxx

```diff
@@ -242,20 +239,63 @@ void TMPWorkerTreeSel::Process(unsigned int code, MPCodeBufPair& msg)
 {
    //evaluate the index of the file to process in fFileNames
    //(we actually don't need the parameter if code == kProcTree)
-   unsigned fileN = 0;
-   unsigned nProcessed = 0;
-   TTree *tree = 0;
+
    Long64_t start = 0;
    Long64_t finish = 0;
+   TEntryList *enl = 0;
+   std::string errmsg;
+   if (LoadTree(code, msg, start, finish, &enl, errmsg) != 0) {
+      SendError(errmsg);
+      return;
+   }
+
+   if (fCallBegin) {
+      fSelector.SlaveBegin(nullptr);
+      fCallBegin = false;
+   }
+
+   fSelector.Init(fTree);
+   fSelector.Notify();
+   for (Long64_t entry = start; entry < finish; ++entry) {
+      Long64_t e = (enl) ? enl->GetEntry(entry) : entry;
+      fSelector.Process(e);
+   }
+
+   // update the number of processed entries
+   fProcessedEntries += finish - start;
+
+   MPSend(GetSocket(), MPCode::kIdling);
+
+   return;
+}
+
+/// Load the requierd tree and evaluate the processing range
+
+Int_t TMPWorkerTree::LoadTree(unsigned int code, MPCodeBufPair &msg, Long64_t &start, Long64_t &finish,
+                              TEntryList **enl, std::string &errmsg)
+{
+   // evaluate the index of the file to process in fFileNames
+   //(we actually don't need the parameter if code == kProcTree)
+
+   start = 0;
+   finish = 0;
+   errmsg = "";
+
+   unsigned fileN = 0;
```

Comment:
please use the full typename unsigned int (Technically, the coding convention says you should use UInt_t ...)